### PR TITLE
Bfla detect findings

### PR DESCRIPTION
**Pull request by danielvladco**

### CODE REVIEW

File: api3/common/openapi.yaml [hunk 1]
```diff
@@ -395,6 +395,12 @@ components:
       type: integer
       format: int64
   parameters:
+    ApiID:
+      name: apiID
+      description: "The path parameter for apiID type"
+      in: path
+      schema:
+        $ref: '#/components/schemas/ApiID'
```
Comment (aduminuc) [inline api3/common/openapi.yaml:403]: We had this before, but for some reason the code generation was failing. I don't know if it's the case, but we decided not to have the ApiID defined as a common parameter section.

File: backend/pkg/modules/internal/bfla/restapi/openapi.yaml [hunk 2]
```diff
@@ -19,6 +19,28 @@ paths:
               schema:
                 $ref: '../../../../../../api3/common/openapi.yaml#/components/schemas/ModuleVersion'
 
+  /api/findings/{apiID}:
+    get:
+      operationId: GetAPIFindingsForAPI
+      summary: 'Get Findings for an API'
+      description: 'Asks APIClarity for all findings of all modules of a specific API. Implemented at the core of APIClarity.'
+      parameters:
+        - $ref: '../../../../../../api3/common/openapi.yaml#/components/parameters/ApiID'
```
Comment (aduminuc) [inline backend/pkg/modules/internal/bfla/restapi/openapi.yaml:28]: As per my previous comment, please use the following:

```
      - name: apiID
          in: path
          required: true
          schema:
            $ref: ../../../../../../api3/common/openapi.yaml#/components/schemas/ApiID
```

File: backend/pkg/modules/internal/bfla/bfladetector/learn_and_detect_bfla.go [hunk 3]
```diff
@@ -57,10 +59,12 @@ func NewBFLADetector(ctx context.Context, apiInfoProvider apiInfoProvider, event
 		apiInfoProvider:          apiInfoProvider,
 		authzModelsMap:           recovery.NewPersistedMap(sp, AuthzModelAnnotationName, reflect.TypeOf(AuthorizationModel{})),
 		tracesCounterMap:         recovery.NewPersistedMap(sp, AuthzProcessedTracesAnnotationName, reflect.TypeOf(1)),
+		findingsMap:              recovery.NewPersistedMap(sp, BFLAFindingsAnnotationName, reflect.TypeOf(1)),
```
Comment (aduminuc) [inline backend/pkg/modules/internal/bfla/bfladetector/learn_and_detect_bfla.go:62]: Make sure to delete them when a reset is received.

File: backend/pkg/modules/internal/bfla/bfladetector/learn_and_detect_bfla.go [hunk 4]
```diff
@@ -422,7 +451,7 @@ func (l *learnAndDetectBFLA) traceRunner(ctx context.Context, trace *CompositeTr
 func (l *learnAndDetectBFLA) notifyController(ctx context.Context, apiID uint) error {
```
Comment (aduminuc) [inline backend/pkg/modules/internal/bfla/bfladetector/learn_and_detect_bfla.go:451]: Are you sending a notification whenever new findings are added and upon reset?

Comment (danielvladco) [reply backend/pkg/modules/internal/bfla/bfladetector/learn_and_detect_bfla.go:455]: yes

File: backend/pkg/modules/internal/bfla/bfladetector/learn_and_detect_bfla.go (hunk 3 above)
Comment (danielvladco) [reply backend/pkg/modules/internal/bfla/bfladetector/learn_and_detect_bfla.go]: good point, fixed

File: backend/pkg/modules/internal/bfla/restapi/openapi.yaml (hunk 2 above)
Comment (danielvladco) [reply backend/pkg/modules/internal/bfla/restapi/openapi.yaml]: fixed

File: api3/common/openapi.yaml (hunk 1 above)
Comment (danielvladco) [reply api3/common/openapi.yaml]: fixed

**Pull request closed without merging**